### PR TITLE
onnx: add Python support to runtime, package onnx-related Python packages

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/onnxruntime/default.nix [hunk 1]
```diff
@@ -37,6 +40,8 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
+  # dontUnpack = true;
```
Comment (ck3d) [inline pkgs/development/libraries/onnxruntime/default.nix:43]: I think this was unintended

**Review by ck3d — APPROVED** [review]
LGTM

File: pkgs/development/libraries/onnxruntime/default.nix [hunk 2]
```diff
@@ -88,12 +102,18 @@ stdenv.mkDerivation rec {
...
+  '';
+
   postInstall = ''
     # perform parts of `tools/ci_build/github/linux/copy_strip_binary.sh`
     install -m644 -Dt $out/include \
       ../include/onnxruntime/core/framework/provider_options.h \
       ../include/onnxruntime/core/providers/cpu/cpu_provider_factory.h \
       ../include/onnxruntime/core/session/onnxruntime_*.h
+  '' + lib.optionalString pythonSupport ''
+    pip install dist/*.whl --no-index --no-warn-script-location --prefix="$python" --no-cache --no-deps
```
Comment (FRidh) [inline pkgs/development/libraries/onnxruntime/default.nix:116]: @jonringer @mweinelt we might be able to use the pipInstallHook here if we make the prefix variable as well as the phase (this is related again to sorted phases).

Comment (mweinelt) [reply pkgs/development/libraries/onnxruntime/default.nix:116]: configurable?

Comment (jonringer) [reply pkgs/development/libraries/onnxruntime/default.nix:116]: sounds good to me, if you can get it to work.

File: pkgs/development/python-modules/skl2onnx/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  # Core dump
+  doCheck = false;
+
+  meta = {
+    description = "Convert scikit-learn models to ONNX";
+    maintainers = with lib.maintainers; [ fridh ];
+    license = with lib.licenses; [ asl20 ];
+  };
+
+
```
Comment (jonringer) [inline pkgs/development/python-modules/skl2onnx/default.nix:50]: NIT: extra newlines
```suggestion
```

**Review by jonringer — COMMENTED** [review]
should we be passing `python3` instead of `python3Packages`?, and just reference `python3.pkgs` for python modules?

**Comment by FRidh** [comment]
Adding Python support is not a good way. Here a wheel is created (which is fine) and installed in a `python` output. The `propagated-build-inputs` file is put in the `dev` output. That's fine, except that `stdenv.mkDerivation` does not automatically add `dev` when `python` is included, whereas it does in case of `out`.

**Pull request merged**